syntax = "proto3";
package identifeye;

message DataPoint {
  string account_name_hash = 1;
  string character_name_hash = 2;
  string ip_hash = 3;
  string ip_location_hash = 4;
  bool is_banned = 5;
  int32 active_playtime = 6;
  string uuid_hash = 7;
}

message BulkData {
  repeated DataPoint data = 1;
}

message Response {
  bool success = 1;
}

service IdentifeyeService {
  rpc SendData (DataPoint) returns (Response);
  rpc SendDataBulk (BulkData) returns (Response);
}